src/bin/flamegraph.rs

```diff
@@ -1,47 +1,43 @@
 use std::path::PathBuf;
 
 use anyhow::anyhow;
-use structopt::StructOpt;
+use clap::{IntoApp, Parser};
+use clap_complete::Shell;
 
 use flamegraph::Workload;
 
-#[derive(Debug, StructOpt)]
-#[structopt(
-    setting = structopt::clap::AppSettings::TrailingVarArg
-)]
+#[derive(Debug, Parser)]
+#[clap(version)]
 struct Opt {
     /// Profile a running process by pid
-    #[structopt(short = "p", long = "pid")]
+    #[clap(short, long)]
     pid: Option<u32>,
 
     /// Generate shell completions for the given shell.
-    #[structopt(long = "completions", conflicts_with = "pid")]
-    completions: Option<structopt::clap::Shell>,
+    #[clap(long, value_name = "SHELL", conflicts_with_all = &["pid", "trailing-arguments"])]
```

Comment:
Why spell out the `value_name` here?

Reply:
It describes what we're actually specifying, i.e. the target shell, otherwise clap just copies the field name